app/js/lib/foxbox/recipes.js

```diff
@@ -24,232 +49,222 @@ export default class Recipe {
-              service.setters = [
-                {
-                  label: 'takes a picture',
-                  value: 'setter:snapshot.ae67e622-7a66-465e-bab0-' +
-                  'b0c5540c5748@link.mozilla.org,TakeSnapshot,null,null'
-                }
-              ];
-              break;
-
-            default:
-              //service.setters = service.setters;
-              break;
-          }
 
-          return service;
+        return {
+          id: getter.id,
+          kind: getter.kind,
+          name: name || getter.adapter,
+          options
+        };
+      });
+    });
+  }
+
+  getSetters() {
+    // Currently we support only Camera and TTS as actuator.
+    const supportedKinds = [
+      'TakeSnapshot',
+      {
+        vendor: 'team@link.mozilla.org',
+        adapter: 'eSpeak adapter',
+        kind: 'Sentence',
+        type: 'String'
+      }
+    ];
+
+    const settersURL = `${this[p.net].origin}/api/` +
+      `v${this[p.settings].apiVersion}/channels/setters`;
+
+    return this[p.net].fetchJSON(
+      settersURL, 'POST', supportedKinds.map(kind => ({ kind }))
+    )
+    .then(setters => {
+      return setters.map(setter => {
+        let name, options = [];
+
+        // Check for the complex extension kind
+        if (typeof setter.kind === 'object') {
+
+          // WORKAROUND: waiting for the Taxonomy fix so that we have "type" for
+          // all places.
+          setter.kind.type = setter.kind.typ;
+          delete setter.kind.typ;
+
+          if (setter.kind.kind === 'Sentence') {
+            name = 'say';
+            options.push(...[{
+              label: '"Good morning!"',
+              value: { String: '"Good morning!"' }
+            }, {
+              label: '"Good afternoon!"',
+              value: { String: '"Good afternoon!"' }
+            }, {
+              label: '"Good evening!"',
+              value: { String: '"Good evening!"' }
+            }]);
+          }
+        } else if (setter.kind === 'TakeSnapshot') {
+          name = 'Camera';
+          options.push({
+              label: 'take a picture',
+              value: { 'Unit': null }
+          });
         }
-      ))
-      // Map all the recipes to a more user-friendly format.
-      .then(services => services.map(service => ({
-        id: service.id,
-        label: service.name,
-        set: service.setters
-      })));
+
+        return {
+          id: setter.id,
+          kind: setter.kind,
+          name: name || setter.adapter,
+          options
+        };
+      });
+    });
   }
 
   /**
    * Create a new recipe.
    *
    * @return {Promise}
    */
-  add(recipe) {
-    const [idGet, kindGet, rangeKeyGet, rangeValGet] = recipe.getter.split(',');
-    const [idSet, kindSet, rangeKeySet, rangeValSet] = recipe.setter.split(',');
-
-    recipe = {
+  add({ recipeName, getter, getterValue, setter, setterValue }) {
+    const recipe = {
+      name: recipeName,
```

Comment:
Can you rename the param `recipeName` to `name` and you can use Object literal property value shorthand.